sky/clouds/vsphere.py

```diff
@@ -0,0 +1,325 @@
+        cluster_name_on_cloud: str,
+        region: 'clouds.Region',
+        zones: Optional[List['clouds.Zone']],
+    ) -> Dict[str, Optional[str]]:
+        # TODO get image id here.
+        assert zones is not None, (region, zones)
+        zone_names = [zone.name for zone in zones]
+        r = resources
+        acc_dict = self.get_accelerators_from_instance_type(r.instance_type)
+        if acc_dict is not None:
+            custom_resources = json.dumps(acc_dict, separators=(',', ':'))
+        else:
+            custom_resources = None
+
+        return {
+            'instance_type': resources.instance_type,
+            'custom_resources': custom_resources,
+            'region': region.name,
+            'zones': ','.join(zone_names),
+        }
+
+    def _get_feasible_launchable_resources(
+            self, resources: 'resources_lib.Resources'):
+        if resources.use_spot:
+            return ([], [])
+        if resources.instance_type is not None:
+            assert resources.is_launchable(), resources
+            resources = resources.copy(accelerators=None)
+            return ([resources], [])
+
+        def _make(instance_list):
+            resource_list = []
+            for instance_type in instance_list:
+                r = resources.copy(
+                    cloud=Vsphere(),
+                    instance_type=instance_type,
+                    accelerators=None,
+                    cpus=None,
+                    memory=None,
+                )
+                resource_list.append(r)
+            return resource_list
+
+        # Currently, handle a filter on accelerators only.
+        accelerators = resources.accelerators
+        if accelerators is None:
+            # Return a default instance type
+            default_instance_type = Vsphere.get_default_instance_type(
+                cpus=resources.cpus,
+                memory=resources.memory,
+                disk_tier=resources.disk_tier,
+            )
+            if default_instance_type is None:
+                return ([], [])
+            else:
+                return (_make([default_instance_type]), [])
+
+        assert len(accelerators) == 1, resources
+        acc, acc_count = list(accelerators.items())[0]
+        (
+            instance_list,
+            fuzzy_candidate_list,
+        ) = service_catalog.get_instance_type_for_accelerator(
+            acc,
+            acc_count,
+            use_spot=resources.use_spot,
+            cpus=resources.cpus,
+            memory=resources.memory,
+            region=resources.region,
+            zone=resources.zone,
+            clouds=_CLOUD_VSPHERE,
+        )
+        if instance_list is None:
+            return ([], fuzzy_candidate_list)
+        return (_make(instance_list), fuzzy_candidate_list)
+
+    @classmethod
+    def check_credentials(cls) -> Tuple[bool, Optional[str]]:
+        """Checks if the user has access credentials to this cloud."""
+
```

Comment:
It would be also good to check if the dependencies are correctly installed in this function, i.e. we should print out some hints if the dependencies are not correctly installed during `sky check`, e.g., https://github.com/skypilot-org/skypilot/blob/9743aa0c7f86a352715b13dd18ee780239f190f6/sky/clouds/aws.py#L464-L486

Reply:
Hints added